### ISSUE TITLE
Improved response validation

### ISSUE THREAD

**Issue by palsivertsen** [eykrehbein/strest#6]
Thanks for this nice tool!

When writing tests it's also important to test failures, e.g. making sure that submitting invalid credentials does not give an auth token. It would be nice if you could do more advanced assertions on the response like so:

```yaml
requests:
  login: # a successful login
    url: http://localhost:8080/auth
    method: POST
    body:
      json:
        username: alice
        password: secret
    response:
      code: [200-300) # expect a response code in range 200 (inclusive) to 300 (exlusive)
      body: 
        type: Base64
  login using params:
    url: http://localhost:8080/auth
    method: POST
    params:
      username: alice
      password: secret
    response:
      code: [200-300) # expect a response code in range 200 (inclusive) to 300 (exlusive)
      body:
        type: Base64
  login wrong password:
    url: http://localhost:8080/auth
    method: POST
    body:
      json:
        username: alice
        password: 123
    response:
      code: [400-500) # expect a response code in range 400 (inclusive) to 500 (exlusive)
      body:
        json:
          error: wrong username or password
```

**Comment by bmcminn** [comment]
I dig the concept of scoping response ranges, however you suggested syntax is not valid YAML. Plus, it's a little esoteric for folks not immediately familiar with mathematical range notations, but documentation can help with that.

Stringifying the value like `code: "[400-500)"` could work better since this is valid YAML and it being of type string tells the config parser it's an range value.

**Comment by eykhagen** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Code validation](https://trello.com/c/hk2lmwBm/10-code-validation)

**Comment by palsivertsen** [comment]
> syntax is not valid YAML
I forgot to validate the file, of course syntax should be valid YAML. Range could have a more basic form: `code: "200-299"`

**Comment by jgroom33** [comment]
resolved with #42
